File: Algorithms/Java/Roman to Integer.java
public class Solution {
    public int romanToInt(String s) {
        int res = 0;
		int flag;
		int[] num = { 1000, 500, 100, 50, 10, 5, 1 };
		String roman = "MDCLXVI";
		for (int i = s.length() - 1; i >= 0; i--) {
			flag = 1;
			char k = s.charAt(i);
			String m = String.valueOf(k);
			if ("CXI".contains(m) && (res >= 5 * num[roman.indexOf(k)])) {
				flag = -1;
			}
			res += (flag * num[roman.indexOf(k)]);
		}
		return res;
    }
}

File: Algorithms/Python/Roman-to-Integer.py
class Solution(object):

    def romanToInt(self, s):
        """
        :type s: str
        :rtype: int
        """
        dic = {'I': 1, 'V': 5, 'X': 10, 'L': 50, 'C': 100, 'D': 500, 'M': 1000}
        ans = 0
        for x in s[::-1]:
            flag = -1 if x in 'IXC' and ans >= 5 * dic[x] else 1
            ans += flag * dic[x]
        return ans
